panel/auth.py

```diff
@@ -1005,7 +1005,7 @@ async def get_user(handler):
             now_ts = dt.datetime.now(dt.timezone.utc).timestamp()
             expiry = None
             if user in state._oauth_user_overrides:
-                while not state._oauth_user_overrides[user]:
+                while not state._oauth_user_overrides.get(user):
```

Comment:
This change isn't safe. If the user override is deleted then the next line after the while loop will error out.

Reply:
Sorry, as you rightly pointed out the problem here is that this change can cause this while loop to run forever.

Reply:
The correct fix is to ensure that we do not delete the user override while it is refreshing:

```python
if state._oauth_user_overrides.get(user) is not None:
    del state._oauth_user_overrides[user]
```

Reply:
Alright it makes sense to me! I reverted the fix changes and applied this one.